contracts/modules/cover/Cover.sol

```diff
@@ -588,6 +556,35 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon {
     return _products.length;
   }
 
+  /*========== COVER EXPIRATION ========== */
+
+  function expireCovers(uint[] calldata coverIds) external {
+    for (uint i = 0; i < coverIds.length; i++) {
+      expireCover(coverIds[i]);
+    }
+  }
+
+  function expireCover(uint coverId) public {
+
+    require(coverAmountTrackingEnabled, "Cover expiring not enabled");
+
+    CoverData memory cover = _coverData[coverId];
+    require(!cover.expired, "Cover: Cover is already expired.");
+
+    uint lastCoverSegmentIndex = _coverSegments[coverId].length - 1;
+    CoverSegment memory lastCoverSegment = _coverSegments[coverId][lastCoverSegmentIndex];
+    if (lastCoverSegment.period + lastCoverSegment.start > block.timestamp) {
+      cover.expired = true;
```

Comment:
Wouldn't it be better to have the expired property on the segment, more precisely on the last segment? That makes it possible for covered vaults to reuse the same NFT even if they decide to not cover it for a certain period of time. It makes things easier to track and implement in UIs and dashboards, and also makes getting cover again on the same product + payout asset pair cheaper for all users since they don't have to mint a new NFT.

Reply:
i thought it would be strange because it's a redundant field in all but the last segment.

In the cover it makes more logical sense